src/Command/Environment/EnvironmentSqlSizeCommand.php

```diff
@@ -0,0 +1,178 @@
+<?php
+namespace Platformsh\Cli\Command\Environment;
+
+use Platformsh\Cli\Command\CommandBase;
+use Platformsh\Cli\Exception\RootNotFoundException;
+use Platformsh\Cli\Helper\ShellHelper;
+use Platformsh\Cli\Local\LocalApplication;
+use Platformsh\Cli\Util\RelationshipsUtil;
+use Platformsh\Cli\Util\Table;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Yaml\Yaml;
+
+class EnvironmentSqlSizeCommand extends CommandBase
+{
+
+    protected function configure()
+    {
+        $this
+            ->setName('environment:sql-size')
+            ->setAliases(['sqls'])
+            ->setDescription('Estimate the disk usage of a database')
+            ->setHelp(
+                "This command provides an estimate of the database's disk usage. It is not guaranteed to be reliable."
+            );
+        $this->addProjectOption()->addEnvironmentOption()->addAppOption();
+        Table::addFormatOption($this->getDefinition());
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->validateInput($input);
+        $projectRoot = $this->getProjectRoot();
+        if (!$projectRoot) {
+            throw new RootNotFoundException();
+        }
+        $appName = $this->selectApp($input);
+
+        $sshUrl = $this->getSelectedEnvironment()
+            ->getSshUrl($appName);
+
+        // Get and parse app config.
+        $app = LocalApplication::getApplication($appName, $projectRoot);
+        $appConfig = $app->getConfig();
+        if (empty($appConfig['relationships'])) {
+            $this->stdErr->writeln('No application relationships found.');
+            return 1;
+        }
+
+        $util = new RelationshipsUtil($this->stdErr);
+        $database = $util->chooseDatabase($sshUrl, $input);
+        if (empty($database)) {
+            $this->stdErr->writeln('No database selected.');
+            return 1;
+        }
+
+        // Find the database's service name in the relationships.
+        $dbServiceName = false;
+        foreach ($appConfig['relationships'] as $relationshipName => $relationship) {
+            if ($database['_relationship_name'] === $relationshipName) {
+                list($dbServiceName,) = explode(':', $relationship, 2);
+                break;
+            }
+        }
+        if (!$dbServiceName) {
+            $this->stdErr->writeln('Service name not found for relationship: ' . $database['_relationship_name']);
+            return 1;
+        }
+
+        // Load services yaml.
+        $services = Yaml::parse(file_get_contents($projectRoot . '/.platform/services.yaml'));
+        $allocatedDisk = $services[$dbServiceName]['disk'];
```

Comment:
This should fail (with `return 1` and a message, or `a RuntimeException`, I don't mind) if the `$services[$dbServiceName]['disk']` is empty. That will save time and potentially save you a division by zero later on.